### PR TITLE
Match some various functions and rework some structs to comply

### PR DESCRIPTION
**Pull request by FazanaJ**
Some renaming's  in order but that can come later.
Changes are getting a little big, hence the now-PR.
Half a draft half not, since there are some known issues stylewise to fix.
Will bring game to 22.67%

### CODE REVIEW

File: src/racer.c [hunk 1]
```diff
@@ -188,15 +189,98 @@ s16 D_8011D5AE;
...
+} Object_64_Unknown5;
+
+typedef struct TempStruct5 {
+    s8 pad0[8];
+    s8 unk8[4][4];
+} TempStruct5;
+
+void func_80043ECC(s32 arg0, Object_64_Unknown5 *arg1, s32 arg2) {
+    TempStruct5 *temp_v0;
+    s8* test;
```
Comment (DavidSM64) [inline src/racer.c:215]:
```suggestion
    s8 *test;
```

File: src/racer.c [hunk 2]
```diff
@@ -463,118 +647,270 @@ void func_80050754(Object *obj, Object_Racer *obj64, f32 arg2) {
...
     arg0->unk16C -= arg0->unk16C >> 3;
     if (arg0->unk16C >= -9 && arg0->unk16C < 10) { // Deadzone?
         arg0->unk16C = 0;
     }
 }
 
-GLOBAL_ASM("asm/non_matchings/racer/func_80052388.s")
+extern s16 func_80070750(f32 y, f32 x);
+
+s32 func_80052388(Object* obj1, Object_Racer* arg1, Object* obj2, f32 distance) {
```
Comment (DavidSM64) [inline src/racer.c:753]:
```suggestion
s32 func_80052388(Object *obj1, Object_Racer *arg1, Object *obj2, f32 distance) {
```

File: src/racer.c [hunk 3]
```diff
@@ -750,16 +1089,73 @@ GLOBAL_ASM("asm/non_matchings/racer/func_80058D5C.s")
...
+    f32 unkA8;
+    u8 padAC[0xE6];
+    s8 unk192;
+    u8 pad196[0x35];
+    u8 unk1C8;
+    u8 pad1C9[3];
+} Object_64_80059080;
+
+void func_80059080(s32 arg0, struct Object_64_80059080 *arg1, f32* arg2, f32* arg3, f32* arg4) {
+    unknown8011AECC* temp_v0_2;
```
Comment (DavidSM64) [inline src/racer.c:1116]:
```suggestion
void func_80059080(s32 arg0, struct Object_64_80059080 *arg1, f32 *arg2, f32 *arg3, f32 *arg4) {
    unknown8011AECC *temp_v0_2;
```

File: src/unknown_003260.c [hunk 4]
```diff
@@ -137,7 +137,38 @@ void audioStopThread(void) {
...
 }
 
-GLOBAL_ASM("asm/non_matchings/unknown_003260/func_80002A98.s")
+/**
+ * Main function for handling ingame audio. Loops continuously as long as the scheduler feeds it updates.
+ */
+void thread4_audio(UNUSED void *arg) {
+    s32 audioThreadMarkExit;
+    s16* audioThreadRetraceMesg;
+    s16* audioThreadUpdateMesg;
```
Comment (DavidSM64) [inline src/unknown_003260.c:146]:
```suggestion
    s16 *audioThreadRetraceMesg;
    s16 *audioThreadUpdateMesg;
```

File: src/object_models.c [hunk 5]
```diff
@@ -67,7 +67,7 @@ void func_8005F850(void) {
         checksum += *(u8 *)(((s32)&func_80024D54) + i);
     }
     if (checksum != gFunc80024D54Checksum) {
```
Comment (Ryan-Myers) [inline src/object_models.c:69]: Can you rename `gFunc80024D54Checksum` to something matching the new function `antipiracy_modify_surface_traction_table`?

**Review by DavidSM64 — COMMENTED** [review]
Asterisk needs to be on the right

File: src/racer.c [hunk 6]
```diff
@@ -463,118 +647,270 @@ void func_80050754(Object *obj, Object_Racer *obj64, f32 arg2) {
...
             }
         }
-    } else {
-        arg0->segment.unk18 = arg3;
-        arg0->segment.unk3B = arg2;
+    } else { // Haven't seen this trigger yet.
+        arg0->segment.unk18.half.upper = arg3;
+        arg0->segment.unk3B = action;
     }
+    /*render_printf("D_8011D55C %d\n", D_8011D55C);
```
Comment (Ryan-Myers) [inline src/racer.c:874]: Just remove these for now.

File: src/racer.c [hunk 7]
```diff
@@ -578,6 +582,7 @@ void obj_init_racer(Object* obj, LevelObjectEntry_CharacterFlag* arg1) {
     tempObj->unk1C6 = 100;
     D_8011D580 = 0;
 
+    // This needs to be on one line to match.
     for (i = 0; i < 4; i++) { D_8011D58C[i] = 0; }
```
Comment (Ryan-Myers) [inline src/racer.c:586]: Just guessing, but that `i< 4` looks like it's just looping through possible players. If so, should it be `i < MAXCONTROLLERS`?

**Pull request merged**